Find a functionally equivalent code:android.widget.ActivityChooserModel.loadActivitiesIfNeeded:COMMENT
Method Modifier: private     
Comment:/**
 * Loads the activities for the current intent if needed which is
 * if they are not already loaded for the current intent.
 *
 * @return Whether loading was performed.
 */

@@ -6,12 +6,15 @@
         final int resolveInfoCount = resolveInfos.size();
         for (int i = 0; i < resolveInfoCount; i++) {
             ResolveInfo resolveInfo = resolveInfos.get(i);
-            mActivities.add(new ActivityResolveInfo(resolveInfo));
+            ActivityInfo activityInfo = resolveInfo.activityInfo;
+            if (ActivityManager.checkComponentPermission(activityInfo.permission, android.os.Process.myUid(), activityInfo.applicationInfo.uid, activityInfo.exported) == PackageManager.PERMISSION_GRANTED) {
+                mActivities.add(new ActivityResolveInfo(resolveInfo));
+            }
         }
         return true;
     }

